### DISCUSSION TITLE
graphql/codegen/docs/integrations/apollo-local-state

### DISCUSSION THREAD

**Discussion by giscus** [dotansimha/graphql-code-generator#10140] [Docs Discussions]
# graphql/codegen/docs/integrations/apollo-local-state

Apollo Local State. Suppose you are using apollo-client >2.5 or older, with apollo-link-state to manage your app state with GraphQL. In that case, you're probably using a client-side only GraphQL schema and client-side directives such as `@client`.

https://the-guild.dev/graphql/codegen/docs/integrations/apollo-local-state

**Comment by agrittiwari** [comment]
what does `If you only plan to perform a query with Apollo’s local state, simply extend the Query type and add the field (query field) you’re trying to call in your local schema:` mean?

I don't write the query/mutation types on client side, Is this a client side step?

**Reply by charpeni** [reply]
It refers to Apollo Client's ability to query against the cache via [local-only fields](https://www.apollographql.com/docs/react/local-state/managing-state-with-field-policies) (aka `@client` directive). If you don't use that feature and don't have any usages of `@client` directive, you can safely skip this step.

When querying local-only fields, GraphQL Code Generator will complain that local-only fields aren't typed as part of the schema because the remote schema (usually) doesn't contain or know about local-only fields.

Therefore, one way to satisfy GraphQL Code Generator is to create a local-only schema besides the remote one. From there, we can extend either `Query` type or other entity types to add proper types for our local-only fields, e.g.:

```graphql
extend type Query {
  todos: Todo
}
```
